donkey.lua

```diff
@@ -27,6 +28,29 @@ end
 local loadSize   = {3, opt.imageSize, opt.imageSize}
 local sampleSize = {3, opt.cropSize, opt.cropSize}
 
+if opt.imgExtInsensitive then
+   -- Overwrite image.load()
+   -- Load an image (with image.load()) regardless its extension
+   function image.load(p, n, t)
+      local img, ok
+      local f = io.open(p, 'r')
+      if f then
+         local h = f:read(4)
+         f:close()
+
+         if h:startswith(string.char(0xff, 0xd8, 0xff)) then
```

Comment:
Yeah, @Atcold found out that it's much faster than what's there at the moment. I can try to add it later on this week